driver-sync/src/test/functional/com/mongodb/internal/connection/OidcAuthenticationProseTests.java

```diff
@@ -268,14 +272,19 @@ private static Stream<Arguments> testValidCallbackInputsTimeoutWhenTimeoutMsIsSe
     }
 
     // Not a prose test
-    @ParameterizedTest(name = "test callback timeout when server selection timeout is "
-            + "infinite and timeoutMs is set to {0}")
-    @ValueSource(ints = {0, 100})
-    void testCallbackTimeoutWhenServerSelectionTimeoutIsInfiniteTimeoutMsIsSet(final int timeoutMs) {
+    @Test
+    @DisplayName("test callback timeout when server selection timeout and "
+            + "timeoutMs are infinite")
+    void testCallbackTimeoutWhenServerSelectionTimeoutIsInfiniteTimeoutMsIsSet() {
         TestCallback callback1 = createCallback();
+        Duration expectedTimeout = ChronoUnit.FOREVER.getDuration();
 
         OidcCallback callback2 = (context) -> {
-            assertEquals(context.getTimeout(), ChronoUnit.FOREVER.getDuration());
+            assertEquals(expectedTimeout, context.getTimeout(),
+                    format("Expected timeout to be infinite (%d seconds), but was %d seconds",
+                            expectedTimeout.getSeconds(),
+                            context.getTimeout().getSeconds()));
```

Comment:
Why are we not just relying on `Duration.toString` here?

```suggestion
            assertEquals(expectedTimeout, context.getTimeout(),
                    format("Expected timeout to be infinite (%s), but was %s",
                            expectedTimeout, context.getTimeout()));
```